### ISSUE TITLE
Use Host header to build urls

### ISSUE THREAD

**Issue by shane-axiom** [azavea/franklin#951]
Currently urls are built using `localhost` unless `--api-host` is explicitly passed. It would be nice if franklin inspected the `Host` header of incoming requests and used the value (if present) as the domain in the constructed url. This would allow a single franklin instance to be hosted under multiple domain names, among other deployment scenarios.

**Comment by shane-axiom** [comment]
Checking [X-Forwarded-Host](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host) would be prudent as well

**Comment by jisantuc** [comment]
Hey @shane-axiom, thanks for the issues! Just a heads up I probably won't be able to get to these until at least two weeks from now since close to 100% of my extra time right now is going toward preparing for a FOSS4G workshop I'm giving. Do you know Scala? If so I'd be happy to a review a PR that makes this change. Also feel free to drop by the Gitter channel if you run into any trouble

**Comment by shane-axiom** [comment]
No worries, I'm not blocked on these particular issues, they're more nice-to-have/best practices kind of requests. Appreciate the response and the project!
